sql/catalog/lease: TestLeaseAtLatestVersion failed

sql/catalog/lease.TestLeaseAtLatestVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18541177?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18541177?buildTab=artifacts#/) on release-25.1 @ [2e1ead314a0fc8acfe06f8c7379932beda5e4f1e](https://github.com/cockroachdb/cockroach/commits/2e1ead314a0fc8acfe06f8c7379932beda5e4f1e):


```
=== RUN   TestLeaseAtLatestVersion
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/5632b52c1c67b61307e1d8a1ca2dd18d/logTestLeaseAtLatestVersion3866879326
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:152: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    lease_test.go:1211: not seeing latest version
    panic.go:626: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5632b52c1c67b61307e1d8a1ca2dd18d/logTestLeaseAtLatestVersion3866879326
--- FAIL: TestLeaseAtLatestVersion (1.66s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseAtLatestVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46620

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -1157,33 +1157,35 @@ func TestLeaseAtLatestVersion(t *testing.T) {
 	srv, sqlDB, kvDB := serverutils.StartServer(t, params)
 	defer srv.Stopper().Stop(context.Background())
 	s := srv.ApplicationLayer()
-
+	// We will intentionally put t2 in a different database, so that wait for
+	// initial version does not apply. Otherwise, a newly published version of kv
+	// will be instantly acquired, since the schema will be leased after the insert
+	// into timestamp.
 	if _, err := sqlDB.Exec(`
 BEGIN;
 CREATE DATABASE t;
+CREATE SCHEMA t.sc1;
 CREATE TABLE t.kv (k CHAR PRIMARY KEY, v CHAR);
-CREATE TABLE t.timestamp (k CHAR PRIMARY KEY, v CHAR);
+CREATE TABLE t.sc1.timestamp (k CHAR PRIMARY KEY, v CHAR);
 INSERT INTO t.kv VALUES ('a', 'b');
 COMMIT;
 `); err != nil {
 		t.Fatal(err)
 	}
-
+	leaseMgr := s.LeaseManager().(*lease.Manager)
 	tableDesc := desctestutils.TestingGetPublicTableDescriptor(kvDB, s.Codec(), "t", "kv")
 	var updated bool
 	if err := crdb.ExecuteTx(context.Background(), sqlDB, nil, func(tx *gosql.Tx) error {
 		// Insert an entry so that the transaction is guaranteed to be
 		// assigned a timestamp.
 		if _, err := tx.Exec(`
-INSERT INTO t.timestamp VALUES ('a', 'b');
+INSERT INTO t.sc1.timestamp VALUES ('a', 'b');
 `); err != nil {
 			return errors.WithStack(err)
 		}
-
 		// Increment the table version after the txn has started. Only do this once
 		// even if there's a retry.
 		if !updated {
-			leaseMgr := s.LeaseManager().(*lease.Manager)
 			if _, err := leaseMgr.Publish(
 				context.Background(), tableDesc.GetID(), func(catalog.MutableDescriptor) error {
 					// Do nothing: increments the version.
@@ -1193,7 +1195,6 @@ INSERT INTO t.timestamp VALUES ('a', 'b');
 			}
 			updated = true
 		}
-
 		// This select will see version 1 of the table. It will first
 		// acquire a lease on version 2 and note that the table descriptor is
 		// invalid for the transaction, so it will read the previous version

